### PR TITLE
Synthesize option utils

### PR DESCRIPTION
**Pull request by m-allanson**
Refs #1715 

This is a fairly lengthy diff, the key part is the addition of [`src/utils/optionsHasKeyword.js`](https://github.com/stylelint/stylelint/pull/1727/files#diff-06350d76302e5d81a8bf17d02224ee7c) and [`src/utils/optionsMatches.js`](https://github.com/stylelint/stylelint/pull/1727/files#diff-50f500900f86f761277b46675a6466e5), along with the removal of the remaining `src/utils/options*.js` files.

### CODE REVIEW

File: src/utils/optionsMatches.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import matchesStringOrRegExp from "./matchesStringOrRegExp"
+
+const extractValueFromInput = (property, input) => {
```
Comment (davidtheclark) [inline src/utils/optionsMatches.js]: I'm in favor of leaving this function out, and simply passing the value into the function directly. Each rule will know exactly how to access the value that it needs to check, and may need some special case; whereas this function locks us into a finite set of properties, making it a little more difficult to add a new one, and will cause problems if/when some rule needs a special case.

Comment (davidtheclark) [inline src/utils/optionsMatches.js]: In fact, if we remove this special treatment of nodes can we possibly consolidate the two utils into one?

Comment (jeddy3) [inline src/utils/optionsMatches.js]: > whereas this function locks us into a finite set of properties, making it a little more difficult to add a new one, and will cause problems if/when some rule needs a special case.

If mitigating this is more of a priority than avoiding code duplication (mainly within the rules using `ignoreAtRules`), then I agree we should simply pass the value in.

@m-allanson Thanks for making a start on this! Can you foresee any issues with either passing in just the value or consolidating the utils? If not, would you mind giving it a go?

Comment (davidtheclark) [inline src/utils/optionsMatches.js]: > If mitigating this is more of a priority than avoiding code duplication

I think so. I think the problem with this approach is that it creates a fragile abstraction that doesn't solve a difficult problem. If have a bunch of rules accessing their value on `node.value`, I don't consider that the bad kind of code duplication.

Comment (jeddy3) [inline src/utils/optionsMatches.js]: >  I think the problem with this approach is that it creates a fragile abstraction that doesn't solve a difficult problem.

Agreed.

Comment (m-allanson) [inline src/utils/optionsMatches.js]: Sounds good, I'll take a look later in the week.

**Comment by m-allanson** [comment]
I have a new PR based on the discussion here. Closing this.

**Pull request closed without merging**